corecfg/proxy.go

```diff
@@ -0,0 +1,74 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package corecfg
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var proxyConfigKeys = map[string]bool{
+	"http":  true,
+	"https": true,
+	"ftp":   true,
+}
+
+func etcEnvironment() string {
+	return filepath.Join(dirs.GlobalRootDir, "/etc/environment")
```

Comment:
Question, why don't we use a file specific to snapd (like an .override file that is applied just to snapd.service). I would not expect a snap config to operate on the whole system. On core / all-snap OS yes, but on classic this is unexpected.